pandas/core/indexes/base.py

```diff
@@ -6150,6 +6153,76 @@ def map(self, mapper, na_action=None):
 
         return Index._with_infer(new_values, dtype=dtype, copy=False, name=self.name)
 
+    def filter(
+        self: _IndexT,
+        items: Axes | None = None,
+        like: str_t | None = None,
+        regex: str_t | None = None,
+    ) -> _IndexT:
+        """
+        Create a subset of the index according to the specified index labels.
+
+        Parameters
+        ----------
+        items : list-like
+            Keep labels from index which are in items.
+        like : str
+            Keep labels from index for which "like in label == True".
+        regex : str (regular expression)
+            Keep labels from index for which re.search(regex, label) == True.
+
+        Returns
+        -------
+        same type as input object
+
+        See Also
+        --------
+        DataFrame.filter : Subset the dataframe rows or columns according to the
+            specified index labels.
+
+        Notes
+        -----
+        The ``items``, ``like``, and ``regex`` parameters are
+        enforced to be mutually exclusive.
+
+        Examples
+        --------
+        >>> idx = pd.Index(["cat", "dog", "bat", "bird"])
+        >>> idx
+        Index(['cat', 'dog', 'bat', 'bird'], dtype='object')
+
+        >>> # select index values by name
+        >>> idx.filter(items=['cat', 'dog'])
+        Index(['cat', 'dog'], dtype='object')
+
+        >>> # select index values by regular expression
+        >>> idx.filter(regex=r'b.*')
+        Index(['bat', 'bird'], dtype='object')
+
+        >>> # select index values containing 'at'
+        >>> idx.filter(like='at')
+        Index(['cat', 'bat'], dtype='object')
+        """
+        nkw = count_not_none(items, like, regex)
+        if nkw > 1:
+            raise TypeError(
+                "Keyword arguments `items`, `like`, or `regex` "
+                "are mutually exclusive"
+            )
+
+        if items is not None:
+            mask = [r in items for r in self]
```

Comment:
This is super slow I guess? You should be able to use isin here.

Reply:
I just copied the pattern that is used in `DataFrame.filter()`  :
https://github.com/pandas-dev/pandas/blob/4d74fbdfbdb8523ed4d73209d8c4c3753da8eb2f/pandas/core/generic.py#L5531